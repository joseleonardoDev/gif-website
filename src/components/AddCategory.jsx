import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewValue }) => {

    const [valueAdded, setValueAdded] = useState('');

    const inputChanged = (event) => {
        setValueAdded(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        const newValue = valueAdded.trim()
        if (newValue.length < 3) return;
        // setCategory(categories =>[...categories, valueAdded])
        setValueAdded('')
        onNewValue(newValue)

    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label="form">
            <input
                type="text"
                placeholder="Search gif"
                value={valueAdded}
                onChange={inputChanged}
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewValue: PropTypes.func.isRequired
}