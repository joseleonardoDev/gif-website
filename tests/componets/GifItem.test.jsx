import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Test in <GifItem/>', () => {

    const title = 'Dragon ball';
    const url = 'https://www.google.com/';
    test('Should be match with the snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })
    test('Should show the image with url and ALT', () => {
        render(<GifItem title={title} url={url} />)
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should show the title', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
    })
});
