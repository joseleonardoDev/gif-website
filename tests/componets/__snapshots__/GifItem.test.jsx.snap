// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> Should be match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://www.google.com/"
    />
    <p>
      Dragon ball
    </p>
  </div>
</div>
`;
